test/test_torch.py

```diff
@@ -5968,23 +5968,33 @@ def test_pow(self, device):
         # [res] torch.pow([res,] x)
 
         # pow has dedicated implementation for different exponents
-        for dtype in torch.testing.get_all_math_dtypes(device):
-
+        dtypes = torch.testing.get_all_math_dtypes(device) + torch.testing.get_all_complex_dtypes()
+        for dtype in dtypes:
             # This test won't work on torch.half because math.pow will generate a much more accurate result. We skip it
             # for now.
             if dtype == torch.half:
                 continue
 
             m1 = torch.empty(0, dtype=dtype, device=device)
-            if m1.is_floating_point():
+            if m1.is_floating_point() or m1.is_complex():
                 m1 = torch.rand(100, 100, dtype=dtype, device=device) + 0.5
             else:
                 # math.pow will overflow and throw exceptions for large integers
                 range_high = 4 if dtype in (torch.int8, torch.uint8) else 10
                 m1 = torch.randint(1, range_high, (100, 100), dtype=dtype, device=device)
 
+            # The complex64 implementation in CPU may show some precision issues
+            # when comparing the resulting float value with python double precision
+            tol_kwargs = {}
+            if dtype in (torch.complex64, torch.float32):
+                tol_kwargs = {'atol': 10e-4}
+
+            exponents = [-2.8, -2, -1, -0.5, 0, 0.5, 1, 2, 3, 4, 3.3]
+            if m1.is_complex():
+                exponents += [-2.5j, -1.0j, 0j, 1.0j, 2.5j, 1.0 + 1.0j, -1.0 - 1.5j]
```

Comment:
you have defined exponents here but I don't see it being used in the code after

Reply:
😅